internal/biz/model/common_representation.go

```diff
@@ -0,0 +1,85 @@
+package model
+
+import (
+	"fmt"
+
+	"github.com/google/uuid"
+)
+
+// CommonRepresentation is an immutable value object representing common resource data.
+// It follows DDD principles where value objects are immutable and should be created
+// through factory methods that enforce validation rules.
+// Note: Fields are exported for GORM compatibility but should not be modified directly.
+type CommonRepresentation struct {
+	BaseRepresentation
+	ResourceId                 uuid.UUID `gorm:"type:text;column:id;primary_key"`
+	ResourceType               string    `gorm:"size:128;column:resource_type"`
+	Version                    uint      `gorm:"type:bigint;column:version;primary_key;check:version > 0"`
+	ReportedByReporterType     string    `gorm:"size:128;column:reported_by_reporter_type"`
```

Comment:
Should we add an index for ReportedByReporterType and ReportedByReporterInstance?

`FindByReporterResourceIdv1beta2` 
`FindByInventoryIdAndReporter ` uses `ReportedByReporterType` and `ReportedByReporterInstance` in the where clause

Reply:
@akoserwal If you look at the proposed algorithm [here](https://docs.google.com/document/d/10FcI3uppt8MJcDKvlHQCQ7h_ujBTw-HvrgEABdxEVbs/edit?tab=t.0#heading=h.v29f3ct0fjjn), we only have one select which will be on resource and representation_reference tables and _not_ on the common_representation and reporter_representation tables.The plan is to deprecate all other read functions as we build out the proposed algorithm since they won't be needed any more. 

A for adding the indexes for read performance, I am thinking we will add them as and when we work on read usecases. This is also the reason why I only focused on unique indexes for these tables.